playbooks/roles/mongo_3_2/templates/backup-mongo.sh.j2

```diff
@@ -0,0 +1,133 @@
+
+# verify required variables are set
+required() {
+    if [ -z "$1" ]; then
+        echo "$2"
+        required_var_missing="yes"
+    fi
+}
+required "$MONGO_BACKUP_EBS_VOLUME_DEVICE" "MONGO_BACKUP_EBS_VOLUME_DEVICE missing; EBS volume device path is required"
+required "$MONGO_BACKUP_EBS_VOLUME_ID" "MONGO_BACKUP_EBS_VOLUME_ID missing; EBS volume id is required"
+required "$MONGO_BACKUP_VOLUME_MOUNT_PATH" "MONGO_BACKUP_VOLUME_MOUNT_PATH missing; path on which to mount ebs backup volume is required"
+required "$MONGO_BACKUP_NODE" "MONGO_BACKUP_NODE missing; this must be set to determine if this is the correct node to run backup on"
+required "$MONGO_BACKUP_AUTH_DATABASE" "MONGO_BACKUP_AUTH_DATABASE missing; this must be set to use the correct authenticationDatabase to auth against"
+required "$MONGO_ADMIN_USER" "MONGO_ADMIN_USER missing; this must be set to auth against the database"
+required "$MONGO_ADMIN_PASSWORD" "MONGO_ADMIN_PASSWORD missing; this must be set to auth against the database"
+required "$AWS_ACCESS_KEY_ID" "MONGO_BACKUP_AWS_ACCESS_KEY_ID missing; this must be set to auth against the database"
+required "$AWS_SECRET_ACCESS_KEY" "MONGO_BACKUP_AWS_SECRET_ACCESS_KEY missing; this must be set to auth against the database"
+required "$aws_region" "aws_region missing; this must be set to use awscli"
+[ -n "$required_var_missing" ] && exit 1
+
+
+# only run on specified node - this pulls the node name (ip address) of the mongo db on this instance
+mynodename=$(echo "db.isMaster()" | mongo -u "$MONGO_ADMIN_USER" -p"$MONGO_ADMIN_PASSWORD" --authenticationDatabase "$MONGO_BACKUP_AUTH_DATABASE" "$EDXAPP_MONGO_DB_NAME" | grep \"me\" | cut -f 2 -d ':' | sed -e 's/ //' -e 's/,//' -e 's/"//');
+if [ "$mynodename" != "$MONGO_BACKUP_NODE" ]; then
+    echo "This is not the backup host. Run on a different instance."
+    exit 1
+fi
+
+# Acquire backup lock using this script itself as the lockfile. If another
+# backup task is already running, then exit immediately.
+exec 200<$0
+flock -n 200 || { echo "Another backup task is already running."; exit 1; }
+
+echo "Starting at $(date)"
+
+# ensure volume is mounted
+mkdir -p "$MONGO_BACKUP_VOLUME_MOUNT_PATH"
+if ! mountpoint -q "$MONGO_BACKUP_VOLUME_MOUNT_PATH"; then
+  mount -o discard,defaults,noatime "$MONGO_BACKUP_EBS_VOLUME_DEVICE" "$MONGO_BACKUP_VOLUME_MOUNT_PATH"
+fi
+
+# Clean old backup files to save space and so we start afresh.
+rm -rf "$MONGO_BACKUP_VOLUME_MOUNT_PATH/mongo/"
+mkdir -p "$MONGO_BACKUP_VOLUME_MOUNT_PATH/mongo/"
+
+
+# create the dump
+# XXX: we may want to check how this lays out of disk and how it will play against ebs volume snapshots. The idea was
+# that snapshots would be cheap because you only pay for data blocks that have changed between one snapshot and the
+# next. however, if the mongodump -> gzip process ends up not being consistent in layout,
+# this may end up with a lot of the disk content changing between snapshots.
```

Comment:
@swalladge Is this still a TODO?

Reply:
@itsjeyd technically, yes. However it's not something we can do much about, nor can we investigate until we have a few weeks of backups. Perhaps we could schedule a ticket to come back to this to investigate?

Reply:
@swalladge Yep, a follow-up ticket would be great 👍 Could you create one and schedule it for Sprint 209?